lib/matplotlib/axis.py

```diff
@@ -2060,13 +2060,26 @@ def _update_offset_text_position(self, bboxes, bboxes2):
         boxes of all the ticklabels
         """
         x, y = self.offsetText.get_position()
-        if not len(bboxes):
-            bottom = self.axes.bbox.ymin
+        if not hasattr(self, '_text_offset_position'):
+            self._text_offset_position = 'bottom'
+        if self._text_offset_position == 'bottom':
+            if not len(bboxes):
+                bottom = self.axes.bbox.ymin
+            else:
+                bbox = mtransforms.Bbox.union(bboxes)
+                bottom = bbox.y0
+            self.offsetText.set_position(
+                (x, bottom - self.OFFSETTEXTPAD * self.figure.dpi / 72)
+            )
+            return
```

Comment:
I find the middle-of-the function return makes the function look less symmetric than it actually is.  I think a `else` block would be clearer to read.